Improving compiler/runtime errors (#698)
realm/realm-kotlin: CHANGELOG.md

### Enhancements
* Improved exception message when attempting to delete frozen objects. (Issue [#616](https://github.com/realm/realm-kotlin/issues/616))
* Added `RealmConfiguration.Builder.compactOnLaunch()`, which can be used to control if a Realm file should be compacted when opened.

> * Improving runtime/compiler errors, (Issue [#684](https://github.com/realm/realm-kotlin/issues/684)), (Issue [#676](https://github.com/realm/realm-kotlin/issues/676)) and , (Issue [#680](https://github.com/realm/realm-kotlin/issues/680)).

Reviewer: ```suggestion
* Data classes can now be used as model classes. Note, it requires default no-arg constructor and properties must still be `var`'s. (Issue [#684](https://github.com/realm/realm-kotlin/issues/684))
* A better error message if Realm plugin was not applied to the module containing model classes. (Issue [#676](https://github.com/realm/realm-kotlin/issues/676))
* A better error message if a class is used that is not part of the schema. (Issue [#680](https://github.com/realm/realm-kotlin/issues/680))
```

This felt a bit too vague...would suggest splitting into multiple lines.

### Enhancements
* Improved exception message when attempting to delete frozen objects. (Issue [#616](https://github.com/realm/realm-kotlin/issues/616))
* Added `RealmConfiguration.Builder.compactOnLaunch()`, which can be used to control if a Realm file should be compacted when opened.

> * A better error message if a Data classes was used as model classes. (Issue [#684](https://github.com/realm/realm-kotlin/issues/684))

Reviewer: ```suggestion
* A better error message if a data class was used as model classes. (Issue [#684](https://github.com/realm/realm-kotlin/issues/684))
```

> * A better error message if Realm plugin was not applied to the module containing model classes. (Issue [#676](https://github.com/realm/realm-kotlin/issues/676))

Reviewer: ```suggestion
* A better error message if the Realm plugin was not applied to the module containing model classes. (Issue [#676](https://github.com/realm/realm-kotlin/issues/676))
```
